### PR TITLE
fail in oracle classic builder if user tries winrm since it doesn't w…

### LINKED ISSUE #5866: Feature request - Oracle OCI build support user-data-file
When you are Building windows server machines using the images Oracle supplies you currently have to log onto them enable winRM/firewalls and other setup feature before creating a custom image that can then be used by packer.  

If the user-data-feature was supported then you would be able to use standard Oracle images sending along a "setup.ps1" file as the user-datafile would then allow us to ensure that all the initial boot configuration could be completed before packer starts provisioning the build.

From the OCI launch help text.
--user-data-file [filename]
---------------------------

A file containing data that Cloud-Init can use to run custom scripts
or provide custom Cloud-Init configuration. This parameter is a
convenience wrapper around the 'user_data' field of the --metadata
parameter.  Populating both values in the same call will result in an
error. For more info see Cloud-Init documentation:
https://cloudinit.readthedocs.org/en/latest/topics/format.html.

### PR DESCRIPTION
**Pull request by SwampDragons**
Add `attributes` and `attributes_file` to the oracle-classic builder. Add validation error if user tries to use winRM.

Closes #5866

### CODE REVIEW

File: builder/oracle/classic/step_create_instance.go [hunk 1]
```diff
@@ -30,13 +32,41 @@ func (s *stepCreateInstance) Run(_ context.Context, state multistep.StateBag) mu
 	// get instances client
 	instanceClient := client.Instances()
 
+	var data map[string]interface{}
```
Comment (mwhooker) [inline builder/oracle/classic/step_create_instance.go]: can we move this to the builder's prepare so we catch data problems early?

File: builder/oracle/classic/step_list_images.go [hunk 2]
```diff
@@ -98,6 +97,29 @@ func (s *stepListImages) Run(_ context.Context, state multistep.StateBag) multis
 	return multistep.ActionContinue
 }
 
+func (s *stepListImages) listForWinRM(state multistep.StateBag) multistep.StepAction {
```
Comment (mwhooker) [inline builder/oracle/classic/step_list_images.go]: imho this is a premature optimization

Comment (mwhooker) [reply builder/oracle/classic/step_list_images.go]: you're so fast you somehow travel back in time

Comment (SwampDragons) [reply builder/oracle/classic/step_list_images.go]: lulz

**Comment by mwhooker** [comment]
👍

**Pull request merged**